includes/Core/Modules/Modules.php

```diff
@@ -639,6 +701,44 @@ private function get_rest_routes() {
 		);
 	}
 
+	/**
+	 * Updates settings for a module.
+	 *
+	 * If the module includes a datapoint 'POST:settings', that datapoint is relied upon, allowing more advanced
+	 * customization. Otherwise this method will ensure every setting is present as a parameter and then update the
+	 * settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Module $module Module to update settings for. Must implement {@see Module_With_Settings}.
+	 * @param array  $data   Associative array of new settings data to set.
+	 * @return bool|WP_Error True on success, or an error object on failure.
+	 */
+	private function update_module_settings( Module $module, array $data ) {
```

Comment:
We actually need to ensure both the class is extended (e.g. it calls `get_datapoints`) _and_ the interface is implemented (for `get_settings`), so I modified the code in a different way to ensure both.

Reply:
That makes sense. I wonder if we should have a `Module_Interface` which the `Module_With_*` interfaces extend it so that we can rely on the single interface to ensure the instance implements all public module methods?

Reply:
Possibly. I'm not sure of the top of my head how that would work in PHP, regarding the following:
* `Module` implements `Module_Interface`
* `Module_With_Settings` extends `Module_Interface`
* `Search_Console` extends `Module` and implements `Module_With_Settings`
* --> does that work in PHP or will there be a "conflict" because theoretically `Module_Interface` is in that hierarchy twice now? (It's not even a conflict, but rather a duplicate.)

Either way, I'd say that's something for a separate issue.

Reply:
Yeah I don't think there's a conflict there, but agreed that it's a separate issue 👍